core/src/main/java/feign/Response.java

```diff
@@ -44,35 +46,150 @@
   private final String reason;
   private final Map<String, Collection<String>> headers;
   private final Body body;
+  private final Request request;
 
+  @Deprecated
```

Comment:
no need to deprecate private methods.. just re-point the factory methods to the builder?